web/src/engine/websites/MangaJar.ts

```diff
@@ -0,0 +1,44 @@
+import { Tags } from '../Tags';
+import icon from './MangaJar.webp';
+import { Chapter, DecoratableMangaScraper, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from './../FetchProvider';
+
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname;
+    const title = anchor.querySelector('p.card-title').textContent.trim();
+    return { id, title };
+}
+
+@Common.MangaCSS(/^https?:\/\/mangajar\.com\/manga\//, 'span.post-name')
+@Common.MangasMultiPageCSS('/manga?page={page}', 'div.row article.flex-item div.post-description a.card-about', 1, 1, 0, MangaInfoExtractor)
+@Common.PagesSinglePageCSS('img.page-image')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangajar', `MangaJar`, 'https://mangajar.com', Tags.Language.English, Tags.Source.Scanlator, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa,);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterslist = [];
+        for (let page = 1, run = true; run; page++) {
+            const chapters = await this.getChaptersFromPage(page, manga);
```

Comment:
Maybe can use a method from existing decorators?

Reply:
Ah you mean, calls to FetchSingleChapterCSS? geez i ma dumb

Reply:
I am afraid, but Common.FetchChaptersSinglePageCSS does not take a path for argument and i didnt find any method that would fit.

I can add an optional argument to that method ofc

`export async function FetchChaptersSinglePageCSS<E extends HTMLElement>(this: MangaScraper, manga: Manga, query: string, extract = DefaultInfoExtractor as InfoExtractor<E>, path : string = ''): Promise<Chapter[]> {
`

Reply:
This might become to confusing, providing a manga and a path at the same time.
Probably best for now to stick with the initial implementation then.

Reply:
In case this use case becomes more general, it might be an option to introduce a new decorator for multi-page chapters, but probably in a separate future PR...